exploration/expression-attributes.md

```diff
@@ -204,38 +217,125 @@ the reserved/private-use rules will need to be adjusted to support attributes.
+  or any other structured formatted results.
+  Ignored when formatting to a string, but could show up in error messages.
+- `u:locale` — A comma-delimited list of BCP 47 language tags,
+  or an implementation-defined list of such tags.
+  The tags are parsed, and they replace the _locale_
+  defined in the _formatting context_ for this expression or markup.
+- `u:dir` — One of the string values `ltr`, `rtl`, or `auto`.
+  Replaces the character directionality
+  defined in the _formatting context_ for this expression or markup.
 
-To distinguish expression attributes from options,
+Error handling should be well defined for invalid values.
+
+Additional restrictions could be imposed,
+e.g. requiring that each `u:id` is unique within a formatted message.
+
+### Attributes
+
+Add support for standalone `@key` as well as
+option-like `@key=value` attribute pairs with a literal value
+at the end of any expression or markup.
+
+To distinguish attributes from options,
 require `@` as a prefix for each attribute asignment.
-Examples: `@translate=yes` and `@locale=$exprLocale`.
+Examples: `@translate=yes` and `@xliff:canCopy`.
 
-Define the meaning and supported values of some expression attributes in the specification,
-including at least `@dir` and `@locale`.
+Do not allow expression or markup attributes to influence the formatting context,
+or pass them to function handlers.
 
-To support later extension of the specified set of attributes while allowing user extensibility,
-suggest custom attribute names to include a U+002D Hyphen-Minus `-`.
-Examples: `@can-copy=no`, `@note-link=|https://...|`.
+Drop variable values from the `attribute` rule:
 
-Allow expression attributes to influence the formatting context,
-but do not directly pass them to user-defined functions.
+```diff
+-attribute = "@" identifier [[s] "=" [s] (literal / variable)]
++attribute = "@" identifier [[s] "=" [s] literal]
+```
 
 ## Alternatives Considered
 
-### Do not support expression attributes
+### Do nothing
+
+Continue to [caution](https://github.com/unicode-org/message-format-wg/blob/d38ff326d2381b3ef361e996c3431d1b251518d6/spec/syntax.md#attributes)
+function authors and other implementers away from creating function-specific or implementation-specific option values
+for the use cases presented above.
+
+As should be obvious, the current situation is not tenable in the long term, and should be resolved.
+
+### Do not provide any guidance
+
+Do not include in the spec rules or guidance for declaring formatted part identifiers,
+or overriding the message locale or directionality.
+
+Do not define a common way to communicate information
+about an expression or markup to translators or tools.
+
+This would mean not defining anything for default registry functions either,
+effectively requiring implementation-specific options like `icu:locale`.
+
+Other functions could use their own definitions and handling for similar options,
+such as `locale` or `x:lang`.
```

Comment:
I think spelling out the pros/cons here might be useful?

> **Pros**
> - implementations would be free to define these options/attributes to suit local implementation needs
> 
> **Cons**
> - users could not count on these options to be available in each function
> - users would have to learn different requirements/restrictions 
>   on option use for each function and for each separate implementation

Reply:
At the moment, these alternative descriptions are not formatted as pro/con lists, and I would rather keep them internally consistent. Therefore, I'd much prefer not adding such a list to just one option, and I don't want to refactor the entire PR again to include such a reformatting. We could do such a reformatting as a follow-on change.